src/qr.jl

```diff
@@ -535,11 +535,11 @@ function ldiv!(A::QRCompactWY{T}, B::AbstractMatrix{T}) where {T}
     return B
 end
 
-function rank(A::QRPivoted; atol::Real=0, rtol::Real=min(size(A)...) * eps(real(float(one(eltype(A.Q))))) * iszero(atol))
+function rank(A::QRPivoted; atol::Real=0, rtol::Real=min(size(A)...) * eps(real(float(eltype(A)))) * iszero(atol))
```

Comment:
Note that `one(eltype(A))` would be better here, in order to handle the case of dimensionful matrices — the `rtol` should always be dimensionless, and `one` strips any units.

Reply:
Sorry, I made that change based on a comment in #1127 . Before making the change I compared `eps(eltype(A))` and `eps(one(eltype(A)))` and verified they give the same answer. I can submit another PR to put back the `one` in there. Also I dont think I understand what you mean by dimensionful here.   Can you point me to a resource to read?